tsdate/core.py

```diff
@@ -1439,32 +1531,48 @@ def main_algorithm(self):
             self.recombination_rate,
             fixed_node_set=self.get_fixed_nodes_set(),
         )
-        return ExpectationPropagation(self.priors, lik, progress=self.pbar)
+        return ExpectationPropagation(
+            self.priors, lik, progress=self.pbar, global_prior=self.global_prior
+        )
 
-    def run(self, eps, max_iterations, max_shape, match_central_moments, global_prior):
+    def run(
+        self,
+        eps,
+        max_iterations,
+        max_shape,
+        match_central_moments,
+        prior_mixture_dim,
+        em_iterations,
+    ):
         if self.provenance_params is not None:
             self.provenance_params.update(
                 {k: v for k, v in locals().items() if k != "self"}
             )
         if not max_iterations >= 1:
             raise ValueError("Maximum number of EP iterations must be greater than 0")
+        if not (isinstance(prior_mixture_dim, int) and prior_mixture_dim > 0):
+            raise ValueError("Number of mixture components must be a positive integer")
         if self.mutation_rate is None:
             raise ValueError("Variational gamma method requires mutation rate")
-        if global_prior:
-            logging.info("Pooling node-specific priors into global prior")
-            self.priors.grid_data[:] = approx.average_gammas(
-                self.priors.grid_data[:, 0], self.priors.grid_data[:, 1]
-            )
+
+        self.global_prior = mixture.initialize_mixture(
```

Comment:
Maybe a comment here about the difference between self.global_prior and self.priors would be helpful?